Makefile.common

```diff
@@ -124,6 +124,8 @@ EXPORT_LDFLAGS := $(LDFLAGS)
 
 # add our coding-style related options
 CFLAGS += -Wall -Wstrict-prototypes -fno-common
+# TODO: try not to break this rule outside of kernel
```

Comment:
maybe add comment regarding WHY this is needed, for example:
`messages are commonly type-punned - avoid GCC optimizations that might break if strict aliasing rules are not adhered to` (or something shorter)

probably the same option should be set in `CXXFLAGS` (compilation flags for C++ programs)